public/app/features/manage-dashboards/utils/validation.ts

```diff
@@ -4,8 +4,29 @@ import { validationSrv } from '../services/ValidationSrv';
 
 export const validateDashboardJson = (json: string) => {
   try {
-    JSON.parse(json);
-    return true;
+    let dashboard = JSON.parse(json);
+    // If dashboard.tags provided && is Array
+    if (dashboard.tags && Array.isArray(dashboard.tags)) {
+      // Check dashboard.tags in Array for only Strings
+      let cont = true;
+      dashboard.tags.forEach((tag: string) => {
+        if (cont === true && typeof tag !== 'string') {
+          cont = false;
+        }
+      });
+      // Continue to import dashboard if Tags in correct format
+      if (cont) {
+        return true;
+      } else {
+        return 'error: tags expected Array of Strings';
+      }
+      // If dashboard.tags && not Array
+    } else if (dashboard.tags && !Array.isArray(dashboard.tags)) {
+      return 'error: tags expected Array';
+      // Else generic JSON test passes and no Tags provided, then import dashboard
+    } else {
+      return true;
+    }
```

Comment:
```suggestion
    let dashboard = JSON.parse(json);
```

Reply:
It seems that there's no need of this code block to be in a `try {} catch {}`. We can just leave the `JSON.parse` then and not return true until all validation has passed?

Reply:
If I'm not wrong, the try/catch is part of the validation. An invalid JSON can be introduced in the input, and this function will be called with it. If we don't have the try/catch the function will throw an uncontrolled error.

Reply:
When an invalid JSON is introduced in the input, the try/catch will catch the error and return `return 'Not valid JSON';`. But the rest of the validation in this method has no need of a try/catch since it's not going to throw unexpected errors from anywhere.